nomo-webon-kit/dist/dart_interface.js

```diff
@@ -62,18 +63,52 @@ const moduleID = Array.from({ length: 8 }, () => String.fromCharCode(Math.floor(
  * This is the main entry point into the native layer.
  */
 export async function invokeNomoFunction(functionName, args) {
+    console.log("invokeNomoFunction", functionName, args);
+    // Handling MetaMask-specific functions
+    if (typeof window.ethereum !== 'undefined') {
+        switch (functionName) {
```

Comment:
please move the fallback-code into individual nomo-functions, like so:

/**
 * Returns blockchain-addresses of the NOMO-user.
 */
export async function nomoGetWalletAddresses(): Promise<{
  walletAddresses: Record<string, string>;
}> {
  if (isFallbackModeActive()) {
   TODO fallback logic